CDP4IME/ShellViewModel.cs

```diff
@@ -91,19 +97,30 @@ public class ShellViewModel : ReactiveObject, IDisposable
         /// </summary>
         private IDisposable subscription;
 
+        /// <summary>
+        /// Gets the <see cref="IAppSettingsService"/> that is used to read app settings
+        /// </summary>
+        private readonly IAppSettingsService appSettingService;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="ShellViewModel"/> class.
         /// </summary>
         /// <param name="dialogNavigationService">
         /// The <see cref="IDialogNavigationService"/> that is used to show modal dialogs to the user
         /// </param>
-        public ShellViewModel(IDialogNavigationService dialogNavigationService)
+        /// <param name="appSettingService">
+        /// The (MEF injected) instance of <see cref="IAppSettingsService"/>
+        /// appSettingService
+        /// </param>
+        public ShellViewModel(IDialogNavigationService dialogNavigationService, IAppSettingsService appSettingService)
         {
             if (dialogNavigationService == null)
             {
                 throw new ArgumentNullException("dialogNavigationService", "The dialogNavigationService may not be null");
             }
 
+            this.appSettingService = appSettingService;
```

Comment:
please add a null check the same way that it is done for the dialogNavigationService

Reply:
Code has been refactored. No needed any longer